File: howmanydigits.cpp
#include <iostream>
#include <unordered_map>
#include <cmath>
using namespace std;

int main() {
    unordered_map<int, int> um;
    um[0] = 1;
    um[1] = 1;
    double result = 0.0;
    for (int i = 2; i < 1000001; i++) {
        result += log10(i);
        um[i] = ceil(result);
    }

    int n;
    while (cin >> n) {
        cout << um[n] << '\n';
    }

    return 0;
}

File: howmanydigits.py
import sys
import math
dict = {0:1, 1:1}
result = 0
for i in range(2, 1000001):
    result += math.log10(i)
    dict[i] = math.ceil(result)

for line in sys.stdin:
    n = int(line.strip())
    print(dict[n])